openstack/resource_openstack_db_database_v1.go

```
@@ -0,0 +1,176 @@
+				Required: true,
+				ForceNew: true,
+			},
+			"instance": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceDatabaseDatabaseV1Create(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	databaseV1Client, err := config.databaseV1Client(GetRegion(d, config))
+	if err != nil {
+		return fmt.Errorf("Error creating cloud database client: %s", err)
+	}
+
+	dbname := d.Get("name").(string)
+
+	var dbs databases.BatchCreateOpts
+	dbs = append(dbs, databases.CreateOpts{
+		Name: dbname,
+	})
+
+	instance_id := d.Get("instance").(string)
+	databases.Create(databaseV1Client, instance_id, dbs)
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"BUILD"},
+		Target:     []string{"ACTIVE"},
+		Refresh:    DatabaseDatabaseV1StateRefreshFunc(databaseV1Client, instance_id, dbname),
+		Timeout:    d.Timeout(schema.TimeoutCreate),
+		Delay:      10 * time.Second,
+		MinTimeout: 3 * time.Second,
+	}
+
+	_, err = stateConf.WaitForState()
+	if err != nil {
+		return fmt.Errorf(
+			"Error waiting for database to become ready: %s", err)
+	}
+
+	// Store the ID now
+	d.SetId(instance_id)
+
+	return resourceDatabaseInstanceV1Read(d, meta)
+}
+
+func resourceDatabaseDatabaseV1Read(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	databaseV1Client, err := config.databaseV1Client(GetRegion(d, config))
+	if err != nil {
+		return fmt.Errorf("Error creating cloud database client: %s", err)
+	}
+
+	dbname := d.Get("name").(string)
+
+	pages, err := databases.List(databaseV1Client, d.Id()).AllPages()
+	if err != nil {
+		return fmt.Errorf("Unable to retrieve databases, pages: %s", err)
+	}
+	allDatabases, err := databases.ExtractDBs(pages)
+	if err != nil {
+		return fmt.Errorf("Unable to retrieve databases, extract: %s", err)
+	}
+
+	for _, v := range allDatabases {
+		if v.Name == dbname {
+			d.Set("name", v.Name)
+			break
+		}
+	}
```

There should be some type of error check here if the dbname could not be found. Similar to what you're doing in `Delete` with `dbExists`. You should return an error if the database could not be found.